### PR TITLE
EVG-15242: Mock HTTP Handler implementation

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-15242

This is a convenience implementation of the `http.Handler` interface for mock HTTP servers. Mostly will be used with `httptest.NewServer`.

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -370,3 +371,55 @@ func RetryHTTPDelay(opts RetryOptions) HTTPDelayFunction {
...
+// NewMockHandler returns a MockHandler object that can be used as an
+// http.Handler.
+func NewMockHandler() *MockHandler { return &MockHandler{} }
+
+// ServerHTTP is a thread-safe handler for mocking HTTP responses. The request
+// URLs are recorded and the customizable header, body, and status code are all
+// written to the http.ResponseWriter. If there is an error when writing to the
+// http.ResponseWriter, a 500 status code is returned to the requester;
+// WriteError (see below) returns the most recent write error.
+func (h *MockHandler) ServerHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (Kimchelly) [inline http.go:397]: Is this supposed to be `ServeHTTP`? Otherwise, it doesn't implement the `http.Handler` interface.

**Review by Kimchelly — CHANGES_REQUESTED** [review]
Since this is a testing utility that will presumably be used in multiple places, is it worth testing that it at least implements the `http.Handler` interface with `assert.Implements` and it also behaves as expected from the doc comment?

File: http.go (hunk 1 above)
Comment (julianedwards) [reply http.go]: Good catch!

File: http.go [hunk 2]
```diff
@@ -407,17 +406,15 @@ func (h *MockHandler) ServerHTTP(w http.ResponseWriter, r *http.Request) {
...
-			return
 		}
 	}
-
-	if h.StatusCode > 0 {
-		w.WriteHeader(h.StatusCode)
-	}
 }
 
 // WriteError returns the most recent error from writing to the
```
Comment (Kimchelly) [inline http.go:420]: Small nit: thinking about it, `WriteError` sounds like you're writing an error and not getting the error that happening during writing, can this be phrased as something else like `GetWriteError`?

**Review by Kimchelly — APPROVED** [review]
LGTM mod small suggestion.

**Pull request merged**